### PR TITLE
original watson bug is fixed by using extension manager when we do info ...

### PR DESCRIPTION
**Pull request by heejaechang**
...bar work.

...

but, Some code clean up.

First is cleanup some infobar code such as adding copyright header, adding internal, removing unused usings and etc.
Second is adding more info in exception log.
Last is making code action more resilient to null return value from user code.

### CODE REVIEW

**Comment by heejaechang** [comment]
@srivatsn @mavasani @shyamnamboodiripad @tmeschter @jmarolf can you take a look please?

**Comment by tmeschter** [comment]
@heejaechang I can't tell what this change is actually about. Could you update the description?

**Comment by heejaechang** [comment]
@tmeschter updated comments

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs [hunk 1]
```diff
@@ -96,10 +96,15 @@ protected static T Cast<T>(object value)
 
         protected static void CheckDeclarationNode<TDeclarationNode>(SyntaxNode destination) where TDeclarationNode : SyntaxNode
         {
+            if (destination == null)
+            {
+                throw new ArgumentNullException("destination");
```
Comment (mavasani) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: nameof(destination)?

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs [hunk 2]
```diff
@@ -96,10 +96,15 @@ protected static T Cast<T>(object value)
...
+            }
+
             if (!(destination is TDeclarationNode))
             {
                 throw new ArgumentException(
-                    string.Format(WorkspacesResources.InvalidDestinationNode, typeof(TDeclarationNode).Name),
+                    string.Format(WorkspacesResources.InvalidDestinationNode, typeof(TDeclarationNode).Name, destination.GetType().Name),
                     "destination");
```
Comment (mavasani) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: here too.. nameof(destination)

**Comment by jmarolf** [comment]
:+1:

File: src/Workspaces/Core/Portable/WorkspacesResources.resx [hunk 3]
```diff
@@ -130,13 +130,13 @@
     <value>Cycle detected in extensions</value>
   </data>
   <data name="InvalidDestinationNode" xml:space="preserve">
-    <value>Destination type must be a {0}.</value>
+    <value>Destination type must be a {0}. but given one is {1}.</value>
```
Comment (mavasani) [inline src/Workspaces/Core/Portable/WorkspacesResources.resx]: remove the "." after the {0}

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs [hunk 4]
```diff
@@ -123,13 +133,18 @@ protected static T Cast<T>(object value)
             where TDeclarationNode2 : SyntaxNode
             where TDeclarationNode3 : SyntaxNode
         {
+            if (destination == null)
```
Comment (mavasani) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: Can you please file a bug to track cleaning up entire CodeGen service to handle null arguments for all public entry points? I remember seeing a few more elsewhere.

**Comment by mavasani** [comment]
:+1:

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs (hunk 4 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: Will do


-          heejae

File: src/EditorFeatures/Core/Implementation/Workspaces/EditorErrorReportingService.cs [hunk 5]
```diff
@@ -1,8 +1,6 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (shyamnamboodiripad) [inline src/EditorFeatures/Core/Implementation/Workspaces/EditorErrorReportingService.cs]: :bulb: We need an analyzer + fixer for missing copyright! :)

Comment (mavasani) [inline src/EditorFeatures/Core/Implementation/Workspaces/EditorErrorReportingService.cs]: File a gitHub issue? :)

Comment (shyamnamboodiripad) [inline src/EditorFeatures/Core/Implementation/Workspaces/EditorErrorReportingService.cs]: #2022!

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs (hunk 1 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: done

File: src/EditorFeatures/Core/Implementation/Workspaces/EditorErrorReportingService.cs (hunk 5 above)
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/Workspaces/EditorErrorReportingService.cs]: :+1:

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs (hunk 2 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: done

File: src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs (hunk 4 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs]: bug opened #2037

File: src/Workspaces/Core/Portable/WorkspacesResources.resx (hunk 3 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/WorkspacesResources.resx]: done

**Pull request merged**